libs/features/mas/web-components/test/merch-card-collection.test.html.js

```diff
@@ -89,18 +89,30 @@ runTests(async () => {
               merchCards.shadowRoot.querySelector('#filtersButton').click();
               await delay(100);
               expect(document.body.classList.contains('merch-modal')).to.be.true;
+              document.querySelector('merch-sidenav').removeAttribute('modal');
+              document.body.classList.remove('merch-modal');
           });
 
-          it('removes the class for modal when closing the filters modal', async () => {
+          it('removes the class for modal when closing the filters modal by clicking the "Close" button', async () => {
+              render();
+              await delay(100);
+              merchCards.shadowRoot.querySelector('#filtersButton').click();
+              await delay(100);
+              document.querySelector('merch-sidenav').shadowRoot.querySelector('#sidenav').querySelector('sp-link').click();
+              expect(document.body.classList.contains('merch-modal')).to.be.false;
+              document.querySelector('merch-sidenav').removeAttribute('modal');
+          });
+
+          it('removes the class for modal when closing the filters modal by clicking outside the modal', async () => {
             render();
             await delay(100);
             merchCards.shadowRoot.querySelector('#filtersButton').click();
             await delay(100);
-            expect(document.body.classList.contains('merch-modal')).to.be.true;
-            document.querySelector('merch-sidenav').shadowRoot.querySelector('#sidenav').querySelector('sp-link').click();
+            document.querySelector('merch-sidenav').shadowRoot.querySelector('sp-overlay').dispatchEvent(new CustomEvent('close'));
```

Comment:
nit: can you simulate a click to a point outside the modal?

Reply:
I tried it, but it doesn't close the modal (only when running unit tests). I think we should cover this not in unit tests, but in international tests (Nala)